src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs

```diff
@@ -56,31 +58,29 @@ public void Execute(ExecuterSettings arguments)
             using (var policyClient = connection.GetClient<PolicyHttpClient>())
             using (var gitRepositoryClient = connection.GetClient<GitHttpClient>())
             {
-                var projects = projectClient.GetProjects();
+                var projects = await projectClient.GetProjects().ConfigureAwait(false);
 
-                foreach (var project in projects.Result)
+                foreach (var project in projects)
                 {
+                    var types = await policyClient.GetPolicyTypesAsync(project.Id).ConfigureAwait(false);
+                    var repositories = await gitRepositoryClient.GetRepositoriesAsync(project.Id).ConfigureAwait(false);
```

Comment:
You should await after the `GetPolicyConfigurationsAsync` call to have them made in parallel